### PR TITLE
Cgo dynamic first test version

### PR DESCRIPTION
**Pull request by alexanderoster**

### CODE REVIEW

File: Source/buildbindinggo.go [hunk 1]
```diff
@@ -396,7 +400,7 @@ func buildGoClass(component ComponentDefinition, class ComponentDefinitionClass,
 	w.Writeln("type %s struct {", class.ClassName)
 	if component.Global.BaseClassName == class.ClassName {
 		w.Writeln("  _     [0]func() // uncomparable; to make == not compile")
-		w.Writeln("  ref   ref       // identifies a C value, see ref type")		
+		w.Writeln("  Ref   ref       // identifies a C value, see ref type")
```
Comment (qmuntal) [inline Source/buildbindinggo.go:403]: @alexanderoster this is an implementation detail and the downstream user should not mess with it, why making it public?

File: Source/buildbindinggo.go [hunk 2]
```diff
@@ -433,12 +437,12 @@ func buildGoClass(component ComponentDefinition, class ComponentDefinitionClass,
 		w.Writeln("")
 		w.Writeln("// Equal reports whether inst and w refer to the same C pointer.")
 		w.Writeln("func (inst %s) Equal(w %s) bool {", class.ClassName, class.ClassName)
-		w.Writeln("  return inst.ref == w.ref")
+		w.Writeln("  return inst.Ref == w.Ref")
 		w.Writeln("}")
 	} else {
-		w.Writeln("func (wrapper * Wrapper) new%s(r ref) %s {", class.ClassName, class.ClassName)
```
Comment (qmuntal) [inline Source/buildbindinggo.go]: @alexanderoster this `new` methods are internal helper functions to be called when instantiating objects inside public methods, but I don't think they should be made public as they do not provide value by their own.

File: Source/buildbindinggo.go [hunk 3]
```diff
@@ -433,12 +437,12 @@ func buildGoClass(component ComponentDefinition, class ComponentDefinitionClass,
...
+		w.Writeln("func (wrapper * Wrapper) New%s(r ref) %s {", class.ClassName, class.ClassName)
 		if class.ParentClass != "" && class.ParentClass != component.Global.BaseClassName {
-			w.Writeln("  return %s{wrapper.new%s(r)}", class.ClassName, class.ParentClass)
+			w.Writeln("  return %s{wrapper.New%s(r)}", class.ClassName, class.ParentClass)
```
Comment (qmuntal) [inline Source/buildbindinggo.go:445]: @alexanderoster this `new` methods are internal helper functions to be called when instantiating objects inside public methods, but I don't think they should be made public as they do not provide value by their own.

File: Source/buildbindinggo.go (hunk 1 above)
Comment (alexanderoster) [reply Source/buildbindinggo.go:403]: Hi @qmuntal, thank you for the hint. I would love to have a better solution to it, but I fear I needed to make the Refs public in order to allow component injection to work. But this is still an open issue. I have merged the different code streams in develop for now in order to keep the branch confusion a bit to a minimum, and I would open the following issues:
- CGo now is using dynamic loading - find a concept to dynamically/statically link the code
- Callbacks are not implemented properly.
- Pointer variables are not implemented properly.
- Component injection works only with slight custom editing of the headers and passing raw Refs around, but this is not a good long-term solution.

Thank you for all your support!

**Pull request merged**